float64 req
---
float64[] res
